src/index.js

```diff
@@ -274,11 +287,19 @@ export default class DavClient {
 	 *
 	 * @param {Array} props
 	 * @param {String} match
+	 * @param {String} test 'anyof', 'allof' or none
 	 * @returns {Promise<Principal[]>}
 	 */
-	async principalPropertySearch(props, match) {
+	async principalPropertySearch(props, match, test) {
 		const [skeleton, propSearch] = XMLUtility.getRootSkeleton(
-			[NS.DAV, 'principal-property-search'], [NS.DAV, 'property-search']);
+			[NS.DAV, 'principal-property-search'],
+			[NS.DAV, 'property-search']
+		);
+		if (test) {
+			skeleton.attributes = [
+				['test', test]
```

Comment:
what?

Reply:
see `\Sabre\DAVACL\AbstractPrincipalCollection::searchPrincipals`. `test` is the argument that tells how the props should be combined, like 'allof' vs 'anyof'

Reply:
`<x0:principal-property-search xmlns:x0="DAV:" test="anyof">…`

Reply:
ok.... I was sure this was just a leftover...